test/res.status.js

```diff
@@ -16,5 +18,42 @@ describe('res', function(){
       .expect('Created')
       .expect(201, done);
     })
+
+    describe('should throw', function() {
+      var InvalidStatusError = new TypeError('Invalid status code')
+
+      it('if status code is < 100 || > 999', function(done) {
+        assert.throws(function() {
+          res.status(99)
```

Comment:
I haven't looked at the implementation yet, but these tests need to be on actual instance objects of res as it would be used by a user; there have been tests done like this in the past and creates lots of issues when internal changes are made that suddenly adds a call / property access that needs the object to be proper. Best just to test it as it is expected to be used.

Reply:
Sounds good, I'll get on it tomorrow. 

CI is also failing and I'll look into that also

Reply:
That's actually why it is failing, though my comment is technically independent from the failing CI; it's just the same issue under the hood, though :)

Reply:
basically you mean every test logic be embedded in

```js
app.use(function (req, res) {
  res.status(..)
});
```

and 

```js
request(app).get('/')..
```

sequences right?

Reply:
Yea, they should use the same pattern as the other tests in this file.